xhistogram/test/fixtures.py

```diff
@@ -12,3 +18,34 @@ def raise_if_computed():
         a = a.rechunk(chunks)
 
     return a
+
+
+@pytest.fixture(scope="module")
```

Comment:
I had always thought that in pytest fixtures were just how you were supposed to handle any input to multiple different test functions. But reading about it more now it seems maybe you only really need fixtures when you need to set up some specific state... Here the dependency is purely functional, so I guess it would be fine to just use functions instead?

Reply:
Typically fixtures are a particularly convenient syntax to use if multiple tests have to run the same setup/cleanup code or if the setup code involves setting up a resource that is shared across multiple tests (pytest fixtures are kind of magical, so I'm sure there are other things they are used for too).

Here, since there's no teardown code, and (I suspect) generating these example datasets is relatively quick, then using plain Python functions should be totally fine.

Reply:
On top of that, your fixtures here weren't generating any data—they were just generating a function. And the same function every time. If you're generating the same immutable thing every time, there's not really any need for a fixture.